bundles/org.eclipse.swt/Eclipse SWT Custom Widgets/common/org/eclipse/swt/custom/CTabFolderRenderer.java

```diff
@@ -355,6 +356,7 @@ protected Point computeSize (int part, int state, GC gc, int wHint, int hHint) {
 							gc.setFont(gcFont);
 						}
 					}
+					width += getTextPadding(item, state);
 					if (parent.showClose || item.showClose) {
 						if ((state & SWT.SELECTED) != 0 || parent.showUnselectedClose) {
 							if (width > 0) width += INTERNAL_SPACING;
```

Comment:
To me, it feels strange if this internal spacing is also applied when no images are shown, as it leads to a large padding right to the text, especially in case the padding has a significantly higher value.

```suggestion
						if (((state & SWT.SELECTED) != 0 && parent.showSelectedImage)
								|| ((state & SWT.SELECTED) == 0 && parent.showUnselectedImage)) {
							if (width > 0) width += INTERNAL_SPACING;
						}
						if ((state & SWT.SELECTED) != 0 || parent.showUnselectedClose) {
```

Reply:
The INTERNAL_SPACING here is applied in case the _close_ is visible for tabs. I believe this has to be there irrespective of the current change.
Also, the spacing is applied **only** when no images are shown. The primary objective of applying the padding is to have proper visual separation between tabs when icons/images are not visible.

I believe there is a gap in my understanding.
@HeikoKlare Could you please elaborate a little in what you wanted to convey via this suggested change?

Reply:
Sorry for the confusion. It's not a gap in your understanding, just something must have went wrong when I added the suggestion. The actual suggestion should be different. Let me explain it textually: The code here does two things: first, it adds the `INTERNAL_SPACING` to the width; seconds, it adds the size of the close button. The latter must _always_ be done, that is correct. But the former one is, from my understanding, supposed to ensure that between the text and the close button there is some spacing. However, when icons are disabled there is already a large gap and adding this addition spacing makes the padding to the left and right of the text uneven.

This is a screenshot with additional spacing (current state):
![image](https://github.com/eclipse-platform/eclipse.platform.swt/assets/755472/b0df66ea-4c92-4e06-b1ce-fa7c53a35a38)

And this without (my proposal):
![image](https://github.com/eclipse-platform/eclipse.platform.swt/assets/755472/091ed3f7-6343-4767-bcdb-d25455db4171)

It is only a small different, but it makes the text look slightly uncentered (because it actually is uncentered).

One note: my explanation is conflicting with your statement:
> Also, the spacing is applied only when no images are shown.

Maybe I missed something in the code, but my understand is that this code will always be excuted when the tab to be drawn has a close icon.

Reply:
@HeikoKlare Thanks for clarifying. The code is now updated accordingly and the tab text is now centered.

`Before`
<img width="450" alt="image" src="https://github.com/eclipse-platform/eclipse.platform.swt/assets/57699330/d8eb1519-d567-449a-b9ae-b37d7d6042d5">

`After`
<img width="450" alt="image" src="https://github.com/eclipse-platform/eclipse.platform.swt/assets/57699330/1b24b568-f283-4fd7-b0fc-868b2bc911f2">
> Also, the spacing is applied only when no images are shown.

>> Maybe I missed something in the code, but my understand is that this code will always be excuted when the tab to be drawn has a close icon.

You are right. The code here gets executed when tab has a close icon. I misunderstood it before.